builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,364 @@
+
+	return multistep.ActionContinue
+}
+
+func (s *StepImportImage) validate(ctx context.Context, logger *PackerLogger) error {
+	logger.Info("Validating image import request...")
+
+	var err error
+	if err = s.isImportFeatureEnabled(ctx, logger); err != nil {
+		return err
+	}
+
+	if err = s.isImportSourceValid(); err != nil {
+		return err
+	}
+
+	if err = s.isImportTargetValid(ctx, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Image import request source and target are valid.")
+	return nil
+}
+
+func (s *StepImportImage) Cleanup(state multistep.StateBag) {
+	if state.Get(StateKeyImageImportRequestCreated) == false {
+		// Either the image import step was skipped or the object was not created successfully.
+		// Skip deleting the ContentLibraryItemImportRequest object.
+		return
+	}
+
+	logger := state.Get("logger").(*PackerLogger)
+	if s.KeepInputArtifact {
+		logger.Info("Skipping clean up of the ContentLibraryItemImportRequest object as specified in config.")
+		return
+	}
+
+	logger.Info(fmt.Sprintf("Deleting the ContentLibraryItemImportRequest object %s in namespace %s.",
+		s.ImportImageConfig.ImportRequestName, s.Namespace))
+	ctx := context.Background()
+	itemImportReqObj := &imgregv1.ContentLibraryItemImportRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ImportImageConfig.ImportRequestName,
+			Namespace: s.Namespace,
+		},
+	}
+	if err := s.KubeWatchClient.Delete(ctx, itemImportReqObj); err != nil {
+		logger.Error("error deleting the ContentLibraryItemImportRequest object : %s", err)
+	} else {
+		logger.Info("Successfully deleted the ContentLibraryItemImportRequest object %s in namespace %s.",
+			s.ImportImageConfig.ImportRequestName, s.Namespace)
+	}
+}
+
+func (s *StepImportImage) initStep(state multistep.StateBag, logger *PackerLogger) error {
+	if err := CheckRequiredStates(state,
+		StateKeySupervisorNamespace,
+		StateKeyKubeClient,
+	); err != nil {
+		logger.Error("error checking required states: %s", err)
+		return err
+	}
+
+	var ok bool
+	if s.Namespace, ok = state.Get(StateKeySupervisorNamespace).(string); !ok {
+		return fmt.Errorf("failed to cast %s to type string", StateKeySupervisorNamespace)
+	}
+	if s.KubeWatchClient, ok = state.Get(StateKeyKubeClient).(client.WithWatch); !ok {
+		return fmt.Errorf("failed to cast %s to type client.WithWatch", StateKeyKubeClient)
+	}
+
+	s.SourceURL = s.ImportImageConfig.ImportSourceURL
+	s.SourceSSLCertificate = s.ImportImageConfig.ImportSourceSSLCertificate
+	s.TargetLocationName = s.ImportImageConfig.ImportTargetLocationName
+	if s.ImportImageConfig.ImportTargetImageType != "" {
+		s.TargetItemType = imgregv1.ContentLibraryItemType(strings.ToUpper(s.ImportImageConfig.ImportTargetImageType))
+	}
+
+	// Use the ImageName from create source config as the target item name.
+	s.TargetItemName = s.CreateSourceConfig.ImageName
+	s.KeepInputArtifact = s.CreateSourceConfig.KeepInputArtifact
```

Comment:
Can we use `TTLSecondsAfterFinished` for the cleanup? Maybe worth exposing an optional config with a default value and passing it when creating the import request CR. I wanted to avoid requiring the `CreateSourceConfig` in this step if possible.

Reply:
I use this `KeepInputArtifact` as a common config, is there a better way to achieve that, or it is not a good practice here? I could use `TTLSecondsAfterFinished`, just feel it would be simpler for users if it can be achieved with a common config for this builder (instead of multiple different configs for different steps). Thoughts?

Reply:
Yes, I agree that a common configuration would be simpler for users, but not sure if that's doable or conventional in custom plugins. Did a quick search in the Packer docs but found nothing related to that. @tenthirtyam do you perhaps know?

Reply:
Yep it works for me. Thanks again for taking care of this to provide a better experience, Eric!

Reply:
Though it works functionally in practice, `make generate` doesn't work and it complains duplicate config names, so it looks it is not supported currently (mapstructure-to-hcl2), I may need to use separate configs instead.